hikari/impl/entity_factory.py

```diff
@@ -1748,6 +1748,20 @@ def _deserialize_command_option(self, payload: data_binding.JSONObject) -> comma
         if raw_channel_types := payload.get("channel_types"):
             channel_types = [channel_models.ChannelType(channel_type) for channel_type in raw_channel_types]
 
+        name_localizations: undefined.UndefinedOr[
+            typing.Mapping[typing.Union[locales.Locale, str], str]
+        ] = undefined.UNDEFINED
+        if raw_name_localizations := payload.get("name_localizations"):
+            items = raw_name_localizations.items()
+            name_localizations = {locales.Locale(key): value for key, value in items}
+
+        description_localizations: undefined.UndefinedOr[
+            typing.Mapping[typing.Union[locales.Locale, str], str]
+        ] = undefined.UNDEFINED
+        if raw_description_localizations := payload.get("description_localizations"):
+            items = raw_description_localizations.items()
+            description_localizations = {locales.Locale(key): value for key, value in items}
```

Comment:
```suggestion
        name_localizations: typing.Mapping[typing.Union[locales.Locale, str], str] = {}
        if raw_name_localizations := payload.get("name_localizations"):
            name_localizations = {locales.Locale(k): raw_name_localizations[k] for k in raw_name_localizations}

        description_localizations: typing.Mapping[typing.Union[locales.Locale, str], str] = {}
        if raw_description_localizations := payload.get("description_localizations"):
            description_localizations = {
                locales.Locale(k): raw_description_localizations[k] for k in raw_description_localizations
            }
```
Doing it this way is significantly faster